androidApp/src/main/java/press/editor/EditorView.kt

```diff
@@ -178,13 +222,37 @@ class EditorView @AssistedInject constructor(
           .popSpan()
           .build()
     }
+
+    onRendered()
   }
 
   private fun render(uiUpdate: EditorUiEffect) {
     when (uiUpdate) {
-      is PopulateContent -> editorEditText.setText(uiUpdate.content, moveCursorToEnd = uiUpdate.moveCursorToEnd)
+      is PopulateContent -> editorEditText.setText(
+          uiUpdate.content, moveCursorToEnd = uiUpdate.moveCursorToEnd
+      )
       is CloseNote -> onDismiss()
     }.exhaustive
+
+    onRendered()
+  }
+
+  private fun onRendered() {
+    val lineDiff = scrollView.getTag(R.id.scrollStateLineDiff)
+    val lineOffset = scrollView.getTag(R.id.scrollStateLineOffset)
+    if (lineDiff is Int && lineOffset is Int) {
+      scrollView.setTag(R.id.scrollStateLineDiff, Any())
+      scrollView.setTag(R.id.scrollStateLineOffset, Any())
+
+      scrollView.doOnNextLayout {
+        scrollView.post {
+          val lineIndex = editorEditText.layout.getLineForOffset(lineOffset)
+          val lineTop = editorEditText.layout.getLineTop(lineIndex)
+          val toScroll = lineTop + lineDiff
+          scrollView.scrollTo(0, toScroll)
```

Comment:
This is impressive. Unfortunately it turns out `TextView` is already capable of doing it on its own as long as it has an ID. This isn't needed.

One problem is that `EditorPresenter#populateNewNotePlaceholderOnStart()` always overrides the text with `# `, throwing away the restored text. `EditorPresenter.Args` could take in a property called `wasRecreated` or something?

Reply:
I see. I did an investigation on TextView but found out that it needs some 'freezeText' setup to let TextView to do the restoration. Also in my test it doesn't correctly restore the line. Will have a double check. Also I realize that my PR has not taken into account the new note scenario (only editing existing note). Will have a 2nd round for this fight :D.

Reply:
Yea `TextView` doesn't save its state without `freezeText`, but `EditText` does. I may want to restore the scroll position when you open an existing note, so this code can come in use sometime in the future. For now, let's let the framework handle its own state.

Reply:
>  Also in my test it doesn't correctly restore the line.

You're right. I'll investigate a bit more if this can be fixed or otherwise, use your implementation.